test/foundry/TokenTransferrer.t.sol

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { BaseConsiderationTest } from "./utils/BaseConsiderationTest.sol";
+import {
+    ConduitTransfer,
+    ConduitBatch1155Transfer,
+    ConduitItemType
+} from "../../contracts/conduit/lib/ConduitStructs.sol";
+import { TestERC20Revert } from "../../contracts/test/TestERC20Revert.sol";
+import { TestERC20NotOk } from "../../contracts/test/TestERC20NotOk.sol";
+import { TestERC721Revert } from "../../contracts/test/TestERC721Revert.sol";
+import { TestERC1155Revert } from "../../contracts/test/TestERC1155Revert.sol";
+import { BaseConduitTest } from "./conduit/BaseConduitTest.sol";
+import { Conduit } from "../../contracts/conduit/Conduit.sol";
+
+contract TokenTransferrerTest is BaseConduitTest {
+    struct Context {
+        Conduit conduit;
+        ConduitTransfer[] transfers;
+        ConduitBatch1155Transfer[] batchTransfers;
+    }
+
+    function test(function(Context memory) external fn, Context memory context)
+        internal
+    {
+        try fn(context) {} catch (bytes memory reason) {
+            assertPass(reason);
+        }
+    }
+
+    function execute(Context memory context) external stateless {
+        context.conduit.execute(context.transfers);
+    }
+
+    function executeBatch(Context memory context) external stateless {
+        context.conduit.executeBatch1155(context.batchTransfers);
+    }
+
+    function testTokenTransferrer() public {
+        ConduitItemType[3] memory itemTypes;
+        itemTypes = [
+            ConduitItemType.ERC20,
+            ConduitItemType.ERC721,
+            ConduitItemType.ERC1155
+        ];
+        ConduitItemType itemType;
+
+        address noCodeTokenAddress;
+        noCodeTokenAddress = address(0xabc);
+
+        uint256 alicePk = 0xa11ce;
```

Comment:
you can use the `forge-std` helper `makeAddr('some string')` to make various addresses; it will also label them with whatever string you provide in forge stack traces - neat!